### ISSUE TITLE
Consolidate the Codewind Che plugins

### ISSUE THREAD

**Issue by johnmcollier** [eclipse/codewind#529]
We currently have two separate Che plugins required for Codewind to function: 
- `codewind-theia` for the Codewind Theia/VS Code extension
- `codewind-sidecar` for the PFE deployment, reverse proxy, and file watching. 

This means that if a user wants to start a Codewind workspace in Che, they need to ensure that both plugins are installed into the workspace, which is usually done by creating a special workspace from one of our [devfiles](https://github.com/eclipse/codewind-che-plugin/tree/master/devfiles). It also means we have to maintain two Che plugins (including an entire container!), it isn't straightforward for a user to install Codewind in an existing workspace, and we don't have a way to easily 'restart' PFE.

Instead, we should look at consolidating the Che plugins into one single Che plugin. This would mean deprecating the `codewind-che-sidecar` container and offloading its responsibilities to the Theia extenstion. This would:
1) Allow us to produce a single Che plugin for Codewind that users can easily install into their Che workspace
    - No need to create special workspaces from devfiles.
    - Codewind could be installed into existing Che workspaces
2) Potentially allow us to onboard Codewind onto the Che plugin registry
3) Consolidate the deployment logic of Codewind
4) Improve error handling when PFE fails to deploy
    - Would be straightforward for the IDE to detect those errors and report back to the user
5) Allow PFE to be redeployed, if needed.

**ToDo:**
- [ ] Use the Codewind installer / CLI to deploy PFE
    - Required for the Hybrid story, can re-use code in https://github.com/eclipse/codewind-che-plugin/tree/master/codewind-che-sidecar/deploy-pfe
    - CLI is already packaged into the Theia container, so this should be doable
    - Need to detect scenarios where PFE can't be deployed or fails to deploy.
- [ ] Use the Typescript filewatcher in Theia
    - Already used in the VS Code extension, so don't have concerns about it working in Theia
- [ ] Deprecate the reverse proxy in the sidecar
   - No longer needed, can get the CLI to pass the PFE service to the Theia extension
- [ ] Do we need to support Theia-less Che workspaces for Hybrid?
   - If so, would need to ensure PFE can still be deployed somehow

**Comment by johnmcollier** [comment]
Marking as a tech-topic since there's a number of architectural changes that need to be discussed as a whole

Also adding `area/vscode-ide` as this will require a few changes to the Theia extension.

**Comment by tetchel** [comment]
Cleaning up issues right now, since this has been put on hold, can you re-add the vs code label or open a new issue once the iterative dev component is finished?
